test/syntax.jl

```diff
@@ -1353,6 +1353,14 @@ end
 # issue #25994
 @test Meta.parse("[a\nfor a in b]") == Expr(:comprehension, Expr(:generator, :a, Expr(:(=), :a, :b)))
 
+# pr 27712
+@test Meta.parse("⟨⟩") == Expr(:anglebracket)
+@test Meta.parse("⟨a,b⟩") == Expr(:anglebracket, :a, :b)
+@test Meta.lower(@__MODULE__, :(⟨a;⟩)) == Expr(:error, "unexpected semicolon inside ⟨ ⟩")
+@test_throws ParseError Meta.parse("⟨a b⟩")
+@test_throws ParseError Meta.parse("⟨1 2;3 4⟩")
+@test Meta.parse("f⟨x⟩") == :(f * $(Expr(:anglebracket, :x)))
```

Comment:
I wonder whether it would make more sense to give this its own expression head, like we already do with `vect`/`ref`, `braces`/`curly`, or `[typed_]h/vcat`, instead of parsing this as juxtaposition.

Reply:
I think that's a good idea.

Reply:
Yeah, `anglecall` probably seems most natural to me.

Reply:
One reason we might want to call it `:anglegetindex` is because it seems natural to also want a `:anglegetindex` for the expression `f⟨x⟩ = 1`, but I don't have a strong feeling either way.

Reply:
Shouldn't it be called `angleref` then? With this PR, your example should still be valid surface syntax for use in macros. I don't think we want to do anything special for this in lowering though, at least for now, since it's not really clear what this should do.